lib/spack/docs/environments.rst

```diff
@@ -274,17 +274,17 @@ Concretizing
 ^^^^^^^^^^^^
 
 Once some user specs have been added to an environment, they can be
-concretized. *By default specs are concretized separately*, one after
-the other. This mode of operation permits to deploy a full
-software stack where multiple configurations of the same package
-need to be installed alongside each other. Central installations done
```

Comment:
This reasoning might be poorly explained:

* We've since added stacks to support this (while that isn't specifically a problem this PR needs to address, this docs section should probably point to the section on stacks for this specific use case)
* Admins might not be the only people who benefit: there may be users who want to install separate tools in a single environment, and it may be that those tools cannot be concretized together (i.e. if each root provides a different top-level executable, but each top-level executable might depend on mutually exclusive library configurations)

Do you think it might be worth adding an option similar to `--with-view`? I think it would be OK to switch the default but IMO we should provide an option so that users can control this from the command line.

Reply:
> This reasoning might be poorly explained:

Sorry to clarify: I mean that the documentation, as originally written, is missing details.